projects/telemetry/inc/nmea.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+// This module is used to parse and represent NMEA data. Used in the GPS module
+
+#include <stdint.h>
+#include <stdio.h>
+#include "status.h"
+
+// This enum contains the list of NMEA sentences (not all of then are supported yet)
+// because they do not contain useful information
+typedef enum {
+  NMEA_UNKNOWN = 0,
+  NMEA_GGA,
+  NMEA_GLL,
+  NMEA_GSA,
+  NMEA_GSV,
+  NMEA_RMC,
+  NMEA_VTG,
+  NUM_NMEA_MESSAGE_ID_TYPES
+} NmeaMessageID;
+
+// An enum to represent a position fix's validity (defined by 1, 2, or 6 as described in
+// https://www.linxtechnologies.com/wp/wp-content/uploads/rxm-gps-f4.pdf
+// on page 14)
+
+typedef enum {
+  NMEA_PF_INVALID = 0,
+  NMEA_PF_GPS_SPS = 1,
+  NMEA_PF_DIFFERENTIAL_GPS_SPS = 2,
+  NMEA_PF_DEAD_RECKONING_MODE = 6,
+  NUM_NMEA_POSITION_FIX_TYPES = 4
+} NmeaPositionFix;
+
+typedef struct {
+  uint8_t hh;    // Hours
+  uint8_t mm;    // Minutes
+  uint8_t ss;    // Seconds
+  uint16_t sss;  // Milliseconds
+} nmea_utc_time;
+
+// Representation of longitude or latitude
+// https://en.wikipedia.org/wiki/Longitude
+typedef struct {
+  uint16_t degrees;
+  uint16_t minutes;
+  uint16_t fraction;
+} nmea_coord;
+
+// Info passed from the GPS chip should be dropped into this struct (more fields
+// coming soon)
+
+typedef struct {
+  NmeaPositionFix position_fix;  // True if this struct has valid data
+  uint16_t satellites_used;
+  uint16_t adc;                 // Age of diff. corr. in seconds
+  uint16_t drs;                 // Diff. Ref. Station. Not sure what it is yet
+  uint16_t hdop_1;              // Horizontal dilution of precision, characteristic
```

Comment:
Please use something more descriptive than `_1` and `_2`.

Reply:
Done